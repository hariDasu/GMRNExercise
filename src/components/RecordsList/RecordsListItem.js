import React from 'react';
import {Platform, StyleSheet, Image, Text, View} from 'react-native';

export const RecordsListItem = ({
  artistName = 'Artist',
  trackName = 'Song Title',
  releaseDate = 'Release Date',
  primaryGenreName = 'Genre',
  trackPrice = '$1.29',
  artWorkUrl = '',
}) => (
  <View style={styles.recordContainer}>
    <View style={styles.imageContainer}>
      <Image style={styles.image} source={{uri: artWorkUrl}} />
    </View>
    <View style={styles.trackDetailsContainer}>
      <Text>
        {artistName} - {trackName}
      </Text>
      <Text>Genre: {primaryGenreName}</Text>
      <Text>Price: {trackPrice}</Text>
      <Text>Released: {releaseDate}</Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  recordContainer: {
    backgroundColor: 'white',
    borderColor: '#d3d3d3',
    borderWidth: 1,
    padding: 10,
    marginTop: 5,
    marginBottom: 5,
    borderRadius: 5,
    width: '90%',
    flexDirection: 'row',
    alignSelf: 'center',
    ...Platform.select({
      android: {
        elevation: 3,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .2)',
        shadowOffset: {height: 0, width: 0},
        shadowOpacity: 3,
        shadowRadius: 2,
      },
    }),
  },
  imageContainer: {
    paddingRight: 10,
  },
  image: {
    height: 100,
    width: 100,
  },
  trackDetailsContainer: {
    flexDirection: 'column',
    flex: 1,
    alignSelf: 'center',
  },
});
