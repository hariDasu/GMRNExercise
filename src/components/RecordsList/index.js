export * from './RecordsFlatList';
export * from './RecordsListItem';
