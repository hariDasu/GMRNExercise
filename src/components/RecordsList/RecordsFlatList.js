import React from 'react';
import {SafeAreaView, FlatList, StyleSheet, StatusBar} from 'react-native';
import {colors} from '../../styles/colors';
import {RecordsListItem} from './RecordsListItem';

export const RecordsList = ({records}) => {
  const renderRecord = ({item}) => {
    const {
      artistName,
      releaseDate,
      trackName,
      primaryGenreName,
      trackPrice,
      artworkUrl100: artWorkUrl,
    } = item;
    return (
      <RecordsListItem
        artistName={artistName}
        releaseDate={releaseDate}
        trackName={trackName}
        primaryGenreName={primaryGenreName}
        trackPrice={trackPrice}
        artWorkUrl={artWorkUrl}
      />
    );
  };

  return (
    <SafeAreaView style={styles.container}>
      <FlatList
        data={records}
        renderItem={renderRecord}
        keyExtractor={(item) => item.trackId.toString()}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: StatusBar.currentHeight || 0,
  },
  item: {
    backgroundColor: colors.primaryBgColor,
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
  },
  title: {
    fontSize: 32,
  },
});
