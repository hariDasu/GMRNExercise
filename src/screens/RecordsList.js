import React, {useState, useCallback} from 'react';
import {StyleSheet, View, TextInput} from 'react-native';
import {GenericButton} from '../components/';
import {useDispatch, useSelector} from 'react-redux';
import * as actions from '../actions';
import {RecordsList as RecordsFlatList} from '../components/RecordsList';

const RecordsList = ({navigation}) => {
  const records = useSelector((state) => state.records.records);
  return (
    <View style={styles.containerStyle}>
      <RecordsFlatList records={records} />
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyle: {
    flex: 1,
  },
});

export default React.memo(RecordsList);
