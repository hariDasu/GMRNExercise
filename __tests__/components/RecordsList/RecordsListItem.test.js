import 'react-native';
import React from 'react';
import {RecordsListItem} from '../../../src/components/RecordsList';

import renderer from 'react-test-renderer';

it('Single RecordsListItem renders correctly across screens', () => {
  const tree = renderer.create(<RecordsListItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
