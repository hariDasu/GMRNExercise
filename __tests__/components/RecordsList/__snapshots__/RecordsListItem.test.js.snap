// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single RecordsListItem renders correctly across screens 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "white",
      "borderColor": "#d3d3d3",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginBottom": 5,
      "marginTop": 5,
      "padding": 10,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 3,
      "shadowRadius": 2,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "paddingRight": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text>
      Artist
       - 
      Song Title
    </Text>
    <Text>
      Genre: 
      Genre
    </Text>
    <Text>
      Price: 
      $1.29
    </Text>
    <Text>
      Released: 
      Release Date
    </Text>
  </View>
</View>
`;
